### DISCUSSION TITLE
[ QUESTION ] ...Expand variable within runscript -Raw

### DISCUSSION THREAD

**Discussion by eddietorial** [CrowdStrike/psfalcon#278] [Q&A] [answered]
I am attempting to wait till xmemdump completes before taking further action on the .dmp file

`$mem_file_path is C:\<yyyy-MM-dd>_<hostname>.dmp`

This works fine as the xmemdump output file in C:\ contains the current date and the hostname that was entered in the main script...
`Invoke-FalconRtr -Command xmemdump -Arguments "Complete $mem_file_path" -HostId $host_id`

However, 

`Invoke-FalconRtr runscript '-Raw=```(Get-ChildItem $mem_file_path).length```' -HostId $host_id).stdout`

just gives me the count of the number of directories in C:\ instead of the size of the $mem_file_path file.

How do I get the $mem_file_path variable to expand within runscript -Raw?

**Comment by bk-cs** [comment, marked as answer]
You are using the wrong type of quotes for what you're trying to do: https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_quoting_rules?view=powershell-7.3

Try this:
`(Invoke-FalconRtr runscript ('-Raw=```(Get-ChildItem {0}).length```' -f $mem_file_path) -HostId $host_id).stdout`
